<?xml version="1.0"?>
<launch>

    <!-- <arg name="start_pt" default= '-7.43, -15.3, 0.8978652, 0.0'/> -->
    <!-- <arg name="start_pt" default= '-7.43, -15.3, 0.0, 0.0'/> -->
    <arg name="start_pt" default= '0.0, 0.0, 0.0, 0.0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="is_sim" default="true"/>
    <arg name="map_file" default="$(find svea_core)/resources/maps/q1.yaml" />
    <arg name="run_lidar" default="true"/>
    <arg name="use_mpc" default="true"/>

    <group if="$(eval use_rviz == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2_team1.rviz" />
    </group>

    <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" />
    <rosparam command="load" file="$(find svea_core)/resources/params/q1obs.yaml" />
    <rosparam command="load" file="$(find svea_core)/resources/params/race_track.yaml" />

    <group if="$(eval use_mpc == false)">
    <node pkg="svea_core" type="floor2_team1.py" name="floor2_team1" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="run_lidar" value="$(arg run_lidar)"/>
    </node>
    </group>

    <group if="$(eval use_mpc == true)">
    <node pkg="svea_core" type="q1_mpc_co_av.py" name="q1_mpc_co_av" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="run_lidar" value="$(arg run_lidar)"/>
    </node>
    </group>

    <!-- <node pkg="gmapping" type="slam_gmapping" name="gmapping" output="screen" > -->
    <!--     <!-1- <remap from="scan" to="scan" /> -1-> -->
    <!--     <remap from="map" to="costmap_node/costmap/costmap" /> -->
    <!--     <param name="odom_frame" value="odom" /> -->
    <!--     <param name="base_frame" value="base_link" /> -->
    <!-- </node> -->

    <node pkg="svea_core" type="tf_sim.py" name="tf_sim" output="screen" />
    <node pkg="tf" type="static_transform_publisher" name="base_link_2_laser" args="0.33 0 0 0 0 0 /base_link /laser 100"/>

    <!-- Publishes the voxel grid to rviz for display -->
    <!-- <node pkg="costmap_2d" type="costmap_2d_markers" name="voxel_visualizer"> -->
    <!--   <remap from="voxel_grid" to="costmap/voxel_grid"/> -->
    <!-- </node> -->

    <!-- Run the costmap node -->
    <node name="costmap_node" pkg="costmap_2d" type="costmap_2d_node" >
      <rosparam file="$(find svea_core)/launch/costmap_params_sim.yaml" command="load" ns="costmap" />
    </node>

    <node pkg="svea_core" type="global_path_node.py" name="global_path_node" output="screen" />    
    <node pkg="svea_core" type="map_explore_node.py" name="explore_node" output="screen" />
    <node pkg="svea_core" type="collision_node.py" name="collision_node" output="screen" />

</launch>
